python/sglang/srt/managers/scheduler_snapshot_system.py

```
@@ -0,0 +1,400 @@
+    # Tier 2: State health monitoring
+    scheduler.state_health_monitor = None
+    scheduler._health_check_counter = {}  # conversation_id → snapshot count
+
+    # Pending-restore registry: canonical rid → loaded snapshot state staged
+    # for the next incoming request. Populated by handle_restore_snapshot when
+    # the originating Req has already finished; consumed by
+    # _maybe_hydrate_from_pending_restore during request creation. Bounded by
+    # PENDING_RESTORE_REGISTRY_MAX on logical (canonical) entries with LRU
+    # eviction. conversation_id aliases are tracked separately in
+    # pending_restore_aliases so replacement and eviction remain atomic.
+    scheduler.pending_restore_registry: "OrderedDict[str, PendingRestoreEntry]" = (
```

Ruff (0.15.13)

[error] Undefined name `PendingRestoreEntry`

(F821)